lib/cursor-position-view.coffee

```diff
@@ -27,18 +29,34 @@ class CursorPositionView extends HTMLElement
   subscribeToConfig: ->
     @configSubscription?.dispose()
     @configSubscription = atom.config.observe 'status-bar.cursorPositionFormat', (value) =>
-      @formatString = value ? '%L:%C'
+      @statusFormat = value ? '%L:%C'
       @updatePosition()
 
   getActiveTextEditor: ->
     atom.workspace.getActiveTextEditor()
 
   updatePosition: ->
-    if position = @getActiveTextEditor()?.getCursorBufferPosition()
-      @row = position.row + 1
-      @column = position.column + 1
-      @textContent = @formatString.replace('%L', @row).replace('%C', @column)
+    if editor = @getActiveTextEditor()
+      screenpos = editor.getCursorScreenPosition()
+      bufpos = editor.getCursorBufferPosition()
+      @row = bufpos.row + 1
+      @column = screenpos.column + 1
+      @lineCount = editor.getLineCount()
+      @length = editor.getText().length
+      @offset = editor.getTextInBufferRange([[0, 1], bufpos]).length
+      @percent = Math.round(100 * @offset / @length)
+      @textContent = @formatString(@statusFormat)
```

Comment:
Actually, [String.replace](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) accepts a function as the replacement. So rather than pre-calculating anything or scanning the string to figure out what to calculate, simply pass the relevant calculation as a function to the replace function.